core/View/SecurityPolicy.php

```diff
@@ -89,4 +103,16 @@ public function createHeaderString() {
 
         return $headerString;
     }
+
+    /**
+     * A less restrictive CSP which will allow embedding other sites with iframes
+     * (useful for heatmaps and session recordings)
+     *
+     */
+    public function allowEmbedPage() {
+        $this->addPolicy('img-src', self::RULE_EMBEDDED_FRAME . ' data:');
+        $this->addPolicy('style-src', self::RULE_EMBEDDED_FRAME);
+        $this->addPolicy('font-src', self::RULE_EMBEDDED_FRAME);
+        $this->addPolicy('frame-src', self::RULE_EMBEDDED_FRAME);
```

Comment:
Looking at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy do we maybe also need to allow `media-src` and `object-src`? not sure.

Also not sure maybe re `style-src-elem` which we may need to allow to load src from other domains? I'm not so much into it.

Otherwise looks good (haven't tested it yet though)

Reply:
@tsteur I have a little better understanding of what we're trying to achieve now and probably the best way to do it is to make the `default-src` looser and then have a more restrictive `script-src` to disallow scripts, and maybe an img-src that allows also `data:` scheme.

1. The embedPage uses a different default-src allowing `http:` `https:`
2. The embedPage has a script-src restricting to the same as the rest of matomo
3. The embed Page has a image-src allowing the same as 1. plus also `data:`